src/GitUrl.js

```diff
@@ -40,20 +40,20 @@ class GitUrl {
       this._path = url.path || defaults.path;
 
       if (!this._owner) {
-        throw Error('Invalid URL: no owner');
+        throw Error('Invalid Git URL: Could not extract owner. Not a github repository url?');
       }
       if (!this._repo) {
-        throw Error('Invalid URL: no repo');
+        throw Error('Invalid Git URL: Could not extract repository. Not a github repository url?');
       }
     } else {
       if (!url) {
-        throw Error('Invalid URL: undefined');
+        throw Error('Invalid URL: Is your repository hosted on github?');
```

Comment:
this message is not very intuitive either. I would write: `Invalid URL: url is undefined.`

Reply:
Hmm, I was getting this error after creating my demo repo and before I uploaded it to github which I was able to get rid of by uploading to gh…hence the error message suggestion

Thinking about how we can make a decent error message for both cli users and users of the Class in js…

Reply:
I think the undefined url comes from another source, which should show the appropriate error. from the point of view of the `GitUrl`, it doesn't know what the application is trying to do. it might not even use github, but our internal git-server.